spec/models/manageiq/providers/azure/cloud_manager/orchestration_stack_spec.rb

```diff
@@ -45,55 +44,117 @@
       end
     end
 
-    context "#update_stack" do
+    describe "#update_stack" do
       it 'updates the stack' do
         expect(orchestration_service).to receive(:create)
-        orchestration_stack.update_stack(template, {})
+        subject.update_stack(template, {})
       end
 
       it 'catches errors from provider' do
         expect(orchestration_service).to receive(:create).and_raise('bad request')
-        expect { orchestration_stack.update_stack(template, {}) }.to raise_error(MiqException::MiqOrchestrationUpdateError)
+        expect { subject.update_stack(template, {}) }.to raise_error(MiqException::MiqOrchestrationUpdateError)
       end
     end
 
-    context "#delete_stack" do
+    describe "#delete_stack" do
       it 'updates the stack' do
         expect(orchestration_service).to receive(:delete)
-        orchestration_stack.delete_stack
+        subject.delete_stack
       end
 
       it 'catches errors from provider' do
         expect(orchestration_service).to receive(:delete).and_raise('bad request')
-        expect { orchestration_stack.delete_stack }.to raise_error(MiqException::MiqOrchestrationDeleteError)
+        expect { subject.delete_stack }.to raise_error(MiqException::MiqOrchestrationDeleteError)
       end
     end
   end
 
   describe 'stack status' do
-    context '#raw_status and #raw_exists' do
-      it 'gets the stack status and reason' do
-        allow(orchestration_service).to receive(:get).and_return(the_raw_stack)
-
-        rstatus = orchestration_stack.raw_status
-        expect(rstatus).to have_attributes(:status => 'Succeeded', :reason => nil)
+    describe '#raw_status and #raw_exists' do
+      context 'stack is created successfully' do
+        before { allow(orchestration_service).to receive(:get).and_return(the_raw_stack) }
+
+        it 'gets the stack status in success' do
+          expect(subject.raw_status).to have_attributes(:status => 'Succeeded', :reason => nil)
+          expect(subject.raw_exists?).to be_truthy
+        end
+      end
 
-        expect(orchestration_stack.raw_exists?).to be_truthy
+      context 'stack creation failed' do
+        before do
+          bad_raw_stack = Azure::Armrest::TemplateDeployment.new('properties' => {'provisioningState' => 'Failed'})
+          operations = [Azure::Armrest::TemplateDeploymentOperation.new('properties' => { 'statusMessage' => 'reason'})]
+          allow(orchestration_service).to receive(:get).and_return(bad_raw_stack)
```

Comment:
@bzwei - A bit of a nick pick but are you testing the scenario where operation.properties.status_message.error.message is populated?